shared/torngit/gitlab.py

```diff
@@ -29,24 +30,242 @@
 METRICS_PREFIX = "services.torngit.gitlab"
 
 
+GITLAB_API_CALL_COUNTER = Counter(
+    "git_provider_api_calls_gitlab",
+    "Number of times gitlab called this endpoint",
+    ["endpoint"],
+)
+
+
+GITLAB_API_ENDPOINTS = {
+    "fetch_and_handle_errors_retry": {
+        "counter": GITLAB_API_CALL_COUNTER.labels(
+            endpoint="fetch_and_handle_errors_retry"
+        ),
+        "url_template": "",  # no url template, just counter
+    },
+    "get_best_effort_branches": {
```

Comment:
[optional] Thoughts on making these keys part of an ENUM? For instance, I've seen "get_best_effort_branches" mentioned 4 times in here, so would centralize the naming if we ever change this. Not a huge pain to find/replace those 4 instances though, whatever you think is better